x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -355,19 +391,21 @@ export class TaskRunner<
       recoveredAlerts: recoveredAlertInstances,
     });
 
-    generateNewAndRecoveredInstanceEvents({
-      eventLogger,
-      originalAlertInstances,
-      currentAlertInstances: instancesWithScheduledActions,
-      recoveredAlertInstances,
-      alertId,
-      alertLabel,
-      namespace,
-      ruleType: alertType,
-      rule: alert,
-    });
+    if (this.shouldLogAndScheduleActionsForAlerts()) {
+      generateNewAndRecoveredInstanceEvents({
+        eventLogger,
+        originalAlertInstances,
+        currentAlertInstances: instancesWithScheduledActions,
+        recoveredAlertInstances,
+        alertId,
+        alertLabel,
+        namespace,
+        ruleType: alertType,
+        rule: alert,
+      });
+    }
 
-    if (!muteAll) {
+    if (!muteAll && this.shouldLogAndScheduleActionsForAlerts()) {
```

Comment:
Hmm. So we create instance events if the rule is muted but not if it's cancelled? That's why we have `!muteAll` here but not in the above conditional?

Reply:
Yes, that's correct. Here's a conversation discussing whether we should be writing out these event log docs if cancelled: https://github.com/elastic/kibana/pull/114289/files#r726163072 (it's a resolved comment so you may have to click to open it?).

Reply:
Okay thanks!